crux-bench/src/crux/bench/watdiv_crux.clj

```diff
@@ -39,22 +41,30 @@
        (into {})))
 
 (defn summarise-query-results [watdiv-query-results]
-  (merge (select-keys (first watdiv-query-results) [:bench-ns :crux-node-type])
-         {:bench-type "queries"
-          :time-taken-ms (->> watdiv-query-results (map :time-taken-ms) (reduce +))}
-         (->> (for [db-name (keys db-query-results)]
-                (let [watdiv-results-with-db (for [query-result watdiv-query-results]
-                                               (merge query-result (get query-result db-name)))
-                      both-completed (->> watdiv-results-with-db (filter (every-pred :db-result-count :result-count)))
-                      crux-correct (->> both-completed (filter #(= (:db-result-count %) (:result-count %))))
-                      correct-idxs (into #{} (map :query-idx) crux-correct)]
-                  [db-name
-                   {:crux-failures (->> both-completed (map :query-idx) (remove correct-idxs) sort vec)
-                    :crux-errors (->> watdiv-results-with-db (filter :db-result-count) (remove :result-count) (map :query-idx) sort vec)
-
-                    :crux-time-taken (->> crux-correct (map :time-taken-ms) (reduce +) (Duration/ofMillis) (str))
-                    :db-time-taken (->> crux-correct (map :db-time-taken-ms) (reduce +) (Duration/ofMillis) (str))}]))
-              (into {}))))
+  (let [base-map (select-keys (first watdiv-query-results) [:bench-ns :crux-node-type])
```

Comment:
just a gut feeling, but this function seems like it's doing a lot - 'transform' and 'push', maybe?

Reply:
Attempt to refactor at a later point.